### PR TITLE
Improve ZLayer how-to documentation

### PR DESCRIPTION
**Pull request by leszekgruchala**

### CODE REVIEW

File: docs/howto/use_modules_and_layers.md [hunk 1]
```diff
@@ -152,13 +153,17 @@ There are many ways to create a `ZLayer`, here's an incomplete list:
...
+ - `ZIO.toLayer` or `ZManaged.toLayer` to construct a layer from an effect
 
 Where it makes sense, these methods have also variants to build a service effectfully (suffixed by `M`), resourcefully (suffixed by `Managed`), or to create a combination of services (suffixed by `Many`).
 
 We can compose `layerA` and `layerB`  _horizontally_ to build a layer that has the requirements of both layers, to provide the capabilities of both layers, through `layerA ++ layerB`
 
-We can also compose layers _vertically_, meaning the output of one layer is used as input for the subsequent layer to build the next layer, resulting in one layer with the requirement of the first and the output of the second layer: `layerA >>> layerB`
+We can also compose layers _vertically_, meaning the output of one layer is used as input for the subsequent layer to build the next layer, resulting in one layer with the requirement of the first, and the output of the second layer: `layerA >>> layerB`.
+When doing this, we must feed all the inputs to make the code compile, but we can still define part of the input as the input of the final layer using `ZLayer.identity`.
```
Comment (adamgfraser) [inline docs/howto/use_modules_and_layers.md:166]:
```suggestion
When doing this, the first layer must output all the services required by the second layer, but we can defer creating some of these services and require them as part of the input of the final layer using `ZLayer.identity`. 
```

File: docs/howto/use_modules_and_layers.md [hunk 2]
```diff
@@ -328,6 +330,18 @@ val layer: ZLayer[Any, Nothing, UserRepo] = connection >>> userRepo
 
 Notice that in `layer` the dependency of `UserRepo` on `Connection` has been "hidden" and is no longer expressed in the type signature. From the perspective of a caller, `layer` just outputs a `UserRepo` and requires no inputs. The caller does not need to be concerned with the internal implementation details of how the `UserRepo` is constructed.
 
+To hide only some inputs we need to explicitly define left overs
```
Comment (adamgfraser) [inline docs/howto/use_modules_and_layers.md:333]:
```suggestion
To provide only some inputs we need to explicitly define what inputs still need to be provided:
```

File: docs/howto/use_modules_and_layers.md [hunk 3]
```diff
@@ -328,6 +330,18 @@ val layer: ZLayer[Any, Nothing, UserRepo] = connection >>> userRepo
...
+```scala mdoc:silent
+trait Configuration
+
+val userRepoWithConfig: ZLayer[Has[Configuration] with Has[Connection], Nothing, UserRepo] = 
+  ZLayer.succeed(new Configuration{}) ++ postgresLayer
+val partialLayer: ZLayer[Has[Configuration], Nothing, UserRepo] = 
+  (ZLayer.identity[Has[Configuration]] ++ connection) >>> userRepoWithConfig
+``` 
+
+In this example the dependency on `Connection` remains hidden, but `Configuration` is still required by `partialLayer`.
```
Comment (adamgfraser) [inline docs/howto/use_modules_and_layers.md:343]:
```suggestion
In this example the requirement for a `Connection` has been satisfied, but `Configuration` is still required by `partialLayer`.
```

**Comment by leszekgruchala** [comment]
@adamgfraser may I ask when do you update the site? I still do not see the changes live from this PR.

**Pull request merged**